### PR TITLE
Issue#493 add authentication to schema registry

### PR DESCRIPTION
**Pull request by MarselAhmetov**
**What changes did you make?** (Give an overview)
I added basic auth for SchemaRegistry and new properties for it 

**How Has This Been Tested?**

- [ ] No need to
- [X] Manually(please, describe, when necessary)
- [ ] Unit checks
- [ ] Integration checks
- [ ] Covered by existing automation

**Checklist**
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (e.g. **ENVIRONMENT VARIABLES**)
- [X] My changes generate no new warnings(e.g. Sonar is happy)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged

### CODE REVIEW

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/config/ClustersProperties.java [hunk 1]
```diff
@@ -35,4 +35,12 @@
     String name;
     String address;
   }
+
+  @Data
+  public static class SchemaRegistry {
+    boolean basicAuthEnabled = false;
+    String username;
+    String password;
+    String address;
```
Comment (germanosin) [inline kafka-ui-api/src/main/java/com/provectus/kafka/ui/config/ClustersProperties.java:44]: url?

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/config/ClustersProperties.java [hunk 2]
```diff
@@ -19,7 +19,7 @@
     String name;
     String bootstrapServers;
     String zookeeper;
-    String schemaRegistry;
+    SchemaRegistry schemaRegistry;
```
Comment (germanosin) [inline kafka-ui-api/src/main/java/com/provectus/kafka/ui/config/ClustersProperties.java:22]: could we support both object and string scenarios?

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/serde/schemaregistry/SchemaRegistryAwareRecordSerDe.java [hunk 3]
```diff
@@ -66,11 +68,25 @@ private static SchemaRegistryClient createSchemaRegistryClient(KafkaCluster clus
     List<SchemaProvider> schemaProviders =
         List.of(new AvroSchemaProvider(), new ProtobufSchemaProvider(), new JsonSchemaProvider());
     //TODO add auth
+
+    Map<String, String> configs = new HashMap<>();
+    if (cluster.getSchemaRegistry().isBasicAuthEnabled()) {
```
Comment (germanosin) [inline kafka-ui-api/src/main/java/com/provectus/kafka/ui/serde/schemaregistry/SchemaRegistryAwareRecordSerDe.java:73]: could we check only user & pass fields? why do we need additional boolean for this?

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/SchemaRegistryService.java [hunk 4]
```diff
@@ -61,7 +64,8 @@
   public Mono<String[]> getAllSubjectNames(String clusterName) {
     return clustersStorage.getClusterByName(clusterName)
         .map(cluster -> webClient.get()
-            .uri(cluster.getSchemaRegistry() + URL_SUBJECTS)
+            .uri(cluster.getSchemaRegistry().getAddress() + URL_SUBJECTS)
+            .headers(headers -> setBasicAuthIfEnabled(cluster, headers))
```
Comment (germanosin) [inline kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/SchemaRegistryService.java:68]: could we set it on webclient level?

**Review by germanosin — CHANGES_REQUESTED** [review]

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/SchemaRegistryService.java (hunk 4 above)
Comment (MarselAhmetov) [reply kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/SchemaRegistryService.java:68]: Here we can make requests for different clusters and in each cluster can be it's own Schema Registry with it's own settings. So we can't use pre configured web client

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/serde/schemaregistry/SchemaRegistryAwareRecordSerDe.java [hunk 5]
```diff
@@ -63,14 +65,29 @@
...
         List.of(new AvroSchemaProvider(), new ProtobufSchemaProvider(), new JsonSchemaProvider());
-    //TODO add auth
+
+    Map<String, String> configs = new HashMap<>();
+    String username = cluster.getSchemaRegistry().getUsername();
+    String password = cluster.getSchemaRegistry().getPassword();
+
+    if (username != null && password != null) {
+      configs.put("basic.auth.credentials.source", "USER_INFO");
+      configs.put("basic.auth.user.info", username + ":" + password);
```
Comment (IldarAlmakaev) [inline kafka-ui-api/src/main/java/com/provectus/kafka/ui/serde/schemaregistry/SchemaRegistryAwareRecordSerDe.java:78]: Could you please move these properties to constants or replace them with existing ones?

**Review by IldarAlmakaev — APPROVED** [review]
Requested a minor change. Overall, looks good to me

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/SchemaRegistryService.java (hunk 4 above)
Comment (MarselAhmetov) [reply kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/SchemaRegistryService.java]: I added method which returns configured WebClient

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/config/ClustersProperties.java (hunk 2 above)
Comment (MarselAhmetov) [reply kafka-ui-api/src/main/java/com/provectus/kafka/ui/config/ClustersProperties.java]: Resolved

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/serde/schemaregistry/SchemaRegistryAwareRecordSerDe.java (hunk 5 above)
Comment (IldarAlmakaev) [reply kafka-ui-api/src/main/java/com/provectus/kafka/ui/serde/schemaregistry/SchemaRegistryAwareRecordSerDe.java:78]: I meant `basic.auth.user.info` and basic.auth.credentials.source

Comment (MarselAhmetov) [reply kafka-ui-api/src/main/java/com/provectus/kafka/ui/serde/schemaregistry/SchemaRegistryAwareRecordSerDe.java]: Done.
I used constants stored in `io.confluent.kafka.serializers.AbstractKafkaSchemaSerDeConfig`.

**Review by germanosin — APPROVED** [review]

**Pull request merged**